gazebo_plugins/include/gazebo_plugins/gazebo_ros_utils.h

```diff
@@ -41,6 +41,14 @@
 #include <gazebo/sensors/Sensor.hh>
 #include <ros/ros.h>
 
+#ifndef GAZEBO_SENSORS_USING_DYNAMIC_POINTER_CAST
+# if GAZEBO_MAJOR_VERSION >= 7
+#define GAZEBO_SENSORS_USING_DYNAMIC_POINTER_CAST using std::dynamic_pointer_cast
```

Comment:
In urdfdom we made [wrapper functions in types.h](https://github.com/ros/urdfdom_headers/blob/master/urdf_model/include/urdf_model/types.h#L72-L95) so you could call `urdf::dynamic_pointer_cast`. In this case, we haven't switched all of gazebo's pointers yet (sensors uses `std::` in gazebo7 while physics still uses `boost::`). It seemed like these `using` macros (which can be scoped) would be a more compact way to allow for different pointer types.